x-pack/solutions/security/plugins/security_solution/server/lib/detection_engine/rule_types/create_security_rule_type_wrapper.ts

```diff
@@ -57,6 +72,19 @@ Object.entries(aadFieldConversion).forEach(([key, value]) => {
   };
 });
 
+const addApmLabelsFromParams = (params: RuleParams) => {
+  agent.addLabels(
```

Comment:
`ruleId` makes sense as something to be indexed, but I'm not sure I understand why we'd want these other fields indexed as well.

Reply:
- `immutable` - necessary to filter for prebuilt rules where the `rule_id` is consistent across clusters
- `from`, `to` - I imagine querying APM data by query range will be interesting to identify rules that are querying large amounts of data. It's also interesting to see if rules are using `to` != `now` and investigate why that might be
- `maxSignals` - since alert creation is a potential CPU bottleneck, filtering for rules with high maxSignals values could help identify behavior patterns when large amounts of alerts are created. Perhaps "created signals" would be more interesting to index instead

Reply:
Agreed that `createdSignals` seems more useful in the abstract; do we capture that anywhere else?

Excellent points in general (that we can probably codify into dashboards) 👍